tests/pypeline/python_cpp_test.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+# Temporary workaround until all the pypeline python classes have defined methods
+# mypy: disable-error-code="attr-defined"
+
+import os
+import sys
+from collections.abc import Buffer
+from ctypes import CDLL
+
+import yaml
+
+from revng.pypeline.analysis import Analysis
+from revng.pypeline.container import Container
+from revng.pypeline.model import Model
+from revng.pypeline.object import Kind, ObjectID
+from revng.pypeline.task.pipe import Pipe
+from revng.pypeline.utils.registry import get_registry, get_singleton
+
+
+def load_extension(paths: list[str]):
+    """Load extension modules via ctypes.CDLL (equivalent to `dlopen`)"""
+    handles = []
+
+    for path in paths:
+        # RTLD_GLOBAL is needed due to nanobind using weak symbols to derive the TypeID
+        handles.append(CDLL(path, os.RTLD_NOW | os.RTLD_GLOBAL))
+
+    import revng.internal._cpp_pypeline as ext
+
+    return (ext, handles)
```

Comment:
Describe why we're dlopening stuff (i.e., loading pipes, containers and analyses) and state what instead is provided by `_cpp_pypeline` (i.e., `Model`, `ObjectID` and `Kind`).

Reply:
Added comments